packages/salesforcedx-vscode-core/src/commands/forceOrgCreate.ts

```diff
@@ -32,13 +39,35 @@ class ForceOrgCreateExecutor extends SfdxCommandletExecutor<FileSelection> {
       )
       .withArg('force:org:create')
       .withFlag('-f', `${selectionPath}`)
+      .withFlag('--setalias', data.alias)
       .withArg('--setdefaultusername')
       .build();
   }
 }
 
+export class AliasGatherer implements ParametersGatherer<Alias> {
+  public async gather(): Promise<CancelResponse | ContinueResponse<Alias>> {
+    const aliasInputOptions = <vscode.InputBoxOptions>{
+      prompt: nls.localize('parameter_gatherer_enter_alias_name')
+    };
+    const alias = await vscode.window.showInputBox(aliasInputOptions);
+    // Hitting enter with no alias will default the alias to 'scratchOrg'
+    if (alias === undefined) {
+      return { type: 'CANCEL' };
+    }
+    return alias === ''
+      ? { type: 'CONTINUE', data: { alias: DEFAULT_ALIAS } }
+      : { type: 'CONTINUE', data: { alias } };
+  }
+}
+export interface Alias {
+  alias: string;
+}
+
 const workspaceChecker = new SfdxWorkspaceChecker();
-const parameterGatherer = new FileSelector('config/**/*-scratch-def.json');
+const parameterGatherer = new CompositeParametersGatherer<
```

Comment:
CompositeParametersGatherer will aggregate all the inputs gathered the type of the Gatherer should be the intersection of the interfaces